packages/core/helper-plugin/src/content-manager/utils/formatContentTypeData.ts

```diff
@@ -5,8 +5,19 @@ import get from 'lodash/get';
 
 import { getOtherInfos, getType } from './getAttributeInfos';
 
-const formatContentTypeData = (data, ct, composSchema) => {
-  const recursiveFormatData = (data, schema) => {
+const formatContentTypeData = <
+  TData extends Record<string, unknown>,
+  TSchema extends Record<string, unknown>,
+  TComponentSchema extends Record<string, object>
+>(
+  data: TData,
+  ct: TSchema,
+  composSchema: TComponentSchema
```

Comment:
Not sure exactly what these are but maybe they could be typed with the strapi [schema](https://github.com/strapi/strapi/blob/main/packages/core/strapi/lib/types/core/schemas/index.d.ts)?